From 8d763281fd5b904821da9e6d189757a67fd0cf1f Mon Sep 17 00:00:00 2001
From: Mohamed Akram <mohd.akram@outlook.com>
Date: Thu, 30 May 2024 22:09:04 +0400
Subject: [PATCH] Fix crash when using pulseaudio

The main loop should be freed only after it has stopped running.
---
 input/pulse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git input/pulse.c input/pulse.c
index 6837a5b..bca8aa2 100644
--- input/pulse.c
+++ input/pulse.c
@@ -26,7 +26,6 @@ void cb(__attribute__((unused)) pa_context *pulseaudio_context, const pa_server_
     pa_context_disconnect(pulseaudio_context);
     pa_context_unref(pulseaudio_context);
     pa_mainloop_quit(m_pulseaudio_mainloop, 0);
-    pa_mainloop_free(m_pulseaudio_mainloop);
 }
 
 void pulseaudio_context_state_callback(pa_context *pulseaudio_context, void *userdata) {
@@ -96,6 +95,7 @@ void getPulseDefaultSink(void *data) {
     }
 
     pa_mainloop_run(m_pulseaudio_mainloop, &ret);
+    pa_mainloop_free(m_pulseaudio_mainloop);
 }
 
 void *input_pulse(void *data) {
